src/App.js

```diff
@@ -1,19 +1,43 @@
 import React from 'react';
+import { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog.js';
+
+const chatDataList = chatMessages
 
 const App = () => {
+  const [chatData, setChatData] = useState(chatDataList);
+
+  const toggleHeart = (id) => {
+    setChatData(chatData => chatData.map(message => {
+      if(message.id === id) {
+        return {...message, liked: !message.liked}
+      } else {
+        return message;
+      }
+    }));
+  }
+
+  const calcHearts = (chatData) => {
+    return chatData.reduce((total, message) => {
+      return total + message.liked;
```

Comment:
Way to use reduce here!

It's cool that you leveraged the knowledge that in JavaScript if you add a boolean to something it gets cast as a number (0 for false, 1 for true). Prefer you use conditional logic to add a number to a number instead, like: 

```javascript
return chat.liked ? total + 1 : total;